### ISSUE TITLE
create restore checkpoint right before every backup

### ISSUE THREAD

**Issue by bithavoc** [wal-g/wal-g#67]
Hi.

I was wondering if it makes sense to create a checkpoint right before each backup, so in case we're using a backup combined with recovery.conf then we can use `recovery_target_name` so it doesn't restore beyond the checkpoint for the backup but it's still able to fetch all the wal logs necessary to reach the point of recovery via `restore_command`.

It should be as easy as calling `pg_create_restore_point('base_<wal-id>')` once we compose the name of wal log for the backup.

This way when you list all backups with `backup-list`, you know you can use the name of the backup as `recovery_target_name` to perform PITR.

Let me know if this makes sense, I can send a PR for it.

**Comment by x4m** [comment]
Johan, thanks for your suggestion!
It looks like recovery_target = 'immediate' will do the trick
https://www.postgresql.org/docs/current/static/recovery-target-settings.html

Though, there is some unexpected useful thing in this: pg_probackup does this to get consistency time of backup.
Look, WAL-G thinks that time of upload of last JSON file - is a minimum possible time for PITR. But it is far from precise. What pg_probackup does is - they call pg_create_restore_point() and scan WAL to look for its first traces with timestamp - and consider this timestamp as a start of available PITR.

This technology requires the insane amount of work and they correctly identify this as "sharp form of paranoia".

Probably, I'm missing something. Any thoghts will be appreciated. Thanks!

**Comment by x4m** [comment]
There is one more problem - WAL-G can create backup from replication slave, where you, probably, cannot call  pg_create_restore_point()

**Comment by bithavoc** [comment]
@x4m `recovery_target = 'immediate'` does the trick indeed.

(hah! when I went through the docs I ignored the first option since it says **"Technically, this is a string parameter, but 'immediate' is currently the only allowed value."** , I thought the parameter was set by default 🤦‍♂️ )

I don't think we need to know when exactly is the recovery time timestamp, what I think it's confusing is that `backup-list` prints `last_modified` that I misinterpreted as a possible parameter for `recovery_target_time`, but this is not the case.

All of this is outside the scope of this ticket, will close now.

Thanks @x4m
